lib/ansible/module_utils/facts/packages.py

```diff
@@ -18,6 +18,8 @@ def get_all_pkg_managers():
 
 class PkgMgr(with_metaclass(ABCMeta, object)):
 
+    warnings = []
```

Comment:
I'm not sure it makes sense to add this is as a class attribute. Warnings aren't inherently something related to a `PkgMgr` object.

Should we create a global value to hold warning state, then process the warning state in `fail_json()` or `exit_json()`?

Reply:
so module.warn?

Reply:
Pretty much. But without the need for a module instance. This would give us a better place to keep warnings that adding a new attribute here.

Reply:
w/o module it cannot be processed in fail_json/exit_json automatically, that is why i did it this way, still works in similar fashion

Reply:
Issue outstanding.

Reply:
Once #58993 is merged in `devel`, we can switch to using those global variables in another PR (that will also need to be backported).

Reply:
@samdoran has fixed this and it will be backported after this and the oother PR which it is blocking is merged https://github.com/ansible/ansible/pull/63476